packages/cli/src/util/build/write-build-result.ts

```diff
@@ -221,10 +234,7 @@ async function writeBuildResultV3(
       })
     : {};
 
-  const ext = extname(src);
-  const path = stripDuplicateSlashes(
-    build.config?.zeroConfig ? src.substring(0, src.length - ext.length) : src
-  );
+  const path = getBuildV3Path(build);
```

Comment:
I'd recommend removing the helper function if it's only being used once here, to keep the diff less noisy.